protocol/testing/e2e/gov/vest_test.go

```diff
@@ -0,0 +1,254 @@
+package gov_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/cometbft/cometbft/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	govtypesv1 "github.com/cosmos/cosmos-sdk/x/gov/types/v1"
+	testapp "github.com/dydxprotocol/v4-chain/protocol/testutil/app"
+	"github.com/dydxprotocol/v4-chain/protocol/testutil/constants"
+	vesttypes "github.com/dydxprotocol/v4-chain/protocol/x/vest/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSetVestEntry(t *testing.T) {
+	tests := map[string]struct {
+		msg                      *vesttypes.MsgSetVestEntry
+		updateExistingVestEntry  bool // whether above msg should update an existing vest entry.
+		expectCheckTxFails       bool
+		expectSubmitProposalFail bool
+		expectedProposalStatus   govtypesv1.ProposalStatus
+	}{
+		"Success: create a new vest entry": {
```

Comment:
do we have a test case with multiple vest entries?

thinking of cases like:
1. multiple vest entries, successfully updates some (i.e. like 1 or 2) entry
2. multiple vest entries, successfully updates all entries

Reply:
oh also 
3. update the same entry multiple times (ie gov msg contains multiple message for the same vest entry)
4. add and then update the same entry (i.e. no entry exists, the gov msg contains 2 msgs, the first msg inserts, then the second msg updates)

Reply:
added more test cases like the ones mentioned here